gio/src/dbus_connection.rs

```diff
@@ -515,4 +725,86 @@ impl DBusConnection {
             );
         }
     }
+
+    // rustdoc-stripper-ignore-next
+    /// Subscribe to a D-Bus signal and receive signal emissions as a stream.
+    ///
+    /// See [`Self::signal_subscribe`] for arguments.  `map_signal` maps the
+    /// received signal to the stream's element.
+    ///
+    /// The returned stream holds a strong reference to this D-Bus connection,
+    /// and unsubscribes from the signal when dropped.
+    ///
+    /// To avoid reference cycles you may wish to downgrade the returned
+    /// stream to hold only weak reference to the connection using
+    /// [`SubscribedSignalStream::downgrade`].
+    pub fn receive_signal<T: 'static, F: Fn(DBusSignalRef) -> T + 'static>(
+        &self,
+        sender: Option<&str>,
+        interface_name: Option<&str>,
+        member: Option<&str>,
+        object_path: Option<&str>,
+        arg0: Option<&str>,
+        flags: DBusSignalFlags,
+        map_signal: F,
```

Comment:
Why this instead of just having the user call e.g. https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html#method.map ?

Reply:
Ah that makes sense, yes

Reply:
I'll leave it as is then.  I guess most of the time folks will use `receive_signal_parameters` anyway, as that's the most common case, I guess.